http/src/main/java/com/github/awsjavakit/http/NewTokenProvider.java

```diff
@@ -0,0 +1,77 @@
+package com.github.awsjavakit.http;
+
+import static com.gtihub.awsjavakit.attempt.Try.attempt;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.awsjavakit.misc.paths.UriWrapper;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Class that fetches a new Bearer token, using an {@link OAuthCredentialsProvider}.
+ */
+public class NewTokenProvider implements TokenProvider {
+
+  public static final String JWT_TOKEN_FIELD = "access_token";
+  public static final String APPLICATION_X_WWW_FORM_URLENCODED = "application/x-www-form-urlencoded";
+  public static final String AUTHORIZATION_HEADER = "Authorization";
+  private static final ObjectMapper JSON = new ObjectMapper();
+  private static final Map<String, String> GRANT_TYPE_CLIENT_CREDENTIALS = Map.of("grant_type",
+    "client_credentials");
+  private static final String CONTENT_TYPE_HEADER = "Content-Type";
+
+  private final HttpClient httpClient;
+  private final OAuthCredentialsProvider credentialsProvider;
+
+  protected NewTokenProvider(HttpClient httpClient,
+    OAuthCredentialsProvider credentialsProvider) {
+    this.httpClient = httpClient;
+    this.credentialsProvider = credentialsProvider;
+  }
+
+  public static NewTokenProvider create(HttpClient httpClient,
+    OAuthCredentialsProvider credentialsProvider){
+    return new NewTokenProvider(httpClient,credentialsProvider);
+  }
+
+  @Override
+  public String fetchToken() {
+    return authenticate();
+  }
+
+  private String authenticate() {
+    var request = formatRequestForOauth2Token();
+    return sendRequestAndExtractToken(request);
+  }
+
+  private HttpRequest formatRequestForOauth2Token() {
+    return HttpRequest.newBuilder(credentialsProvider.getAuthorizationEndpoint())
+      .header(AUTHORIZATION_HEADER, credentialsProvider.getAuthorizationHeader())
+      .header(CONTENT_TYPE_HEADER, APPLICATION_X_WWW_FORM_URLENCODED)
+      .POST(clientCredentialsAuthType())
+      .build();
+  }
+
+  private HttpRequest.BodyPublisher clientCredentialsAuthType() {
+    var queryParameters = UriWrapper.fromHost("notimportant")
```

Comment:
While this is a bit of a weird way of doing this I understand why, "not important" here should be a constant that explains that that the URI is a dummy.